azure-mgmt-resources/src/main/java/com/microsoft/azure/management/resources/implementation/ResourceGroupsImpl.java

```diff
@@ -37,6 +37,21 @@
     }
 
     @Override
+    public PagedList<ResourceGroup> listByTag(String tagName, String tagValue) {
+        if (tagName == null) {
+            throw new IllegalArgumentException("tagName == null");
+        }
+        String odataFilter;
```

Comment:
As you are splitting the code for listing at subscription and at resource group, I assume generating ODataFilter is common code at both locations. If that is the case can you create a utility to create the ODataFilter string.